lib/puppet/provider/login.rb

```diff
@@ -19,43 +19,79 @@
 # The three possible ways of declaring the variables are, respectively:
 def login
+  options = {
+    type:         'OneView',
+    file_env_var: 'ONEVIEW_AUTH_FILE',
+    env_var_url:  'ONEVIEW_URL',
+    filename:     '/login.json'
+  }
+  credentials = load_authentication_settings(options)
+  credentials[:hardware_variant] ||= 'C7000'
+  credentials
+end
+
+# This method returns the information necessary in order to log in to the Image Streamer Appliance
+# The three possible ways of declaring the variables are, respectively:
+def login_image_streamer
+  options = {
+    type:         'Image Streamer',
+    file_env_var: 'IMAGE_STREAMER_AUTH_FILE',
+    env_var_url:  'IMAGE_STREAMER_URL',
+    filename:     '/login_image_streamer.json'
+  }
+  load_authentication_settings(options)
+end
+
+def load_authentication_settings(options)
   begin
-    # - Creating a JSON file with the proper fields and setting the environment variable ONEVIEW_AUTH_FILE to its path
-    credentials = if ENV['ONEVIEW_AUTH_FILE']
-                    JSON.parse(File.read(File.absolute_path(ENV['ONEVIEW_AUTH_FILE'])), symbolize_names: true)
+    # - Creating a JSON file with the proper fields and setting the environment variable with an auth file to its path
+    credentials = if ENV[options[:file_env_var]]
+                    JSON.parse(File.read(File.absolute_path(ENV[options[:file_env_var]])), symbolize_names: true)
                   # - Declaring each field as an environment variable
-                  elsif ENV['ONEVIEW_URL']
-                    environment_credentials
+                  elsif ENV[options[:env_var_url]]
+                    environment_credentials(options)
                   # - Placing a JSON file in the directory you are running the manifests from
                   else
-                    JSON.parse(File.read(File.expand_path(Dir.pwd + '/login.json', __FILE__)), symbolize_names: true)
+                    JSON.parse(File.read(File.expand_path(Dir.pwd + options[:filename], __FILE__)), symbolize_names: true)
                   end
   rescue
-    raise('The Oneview credentials could not be set. Please check the documentation for more information.')
+    raise "The #{options[:type]} credentials could not be set. Please check the documentation for more information."
   end
   credentials_parse(credentials)
 end
 
 # Returns the credentials set by environment variables
-def environment_credentials
+def environment_credentials(options)
+  options[:type] == 'OneView' ? environment_credentials_from_oneview : environment_credentials_from_image_streamer
+end
+
+def environment_credentials_from_oneview
   {
     url:                     ENV['ONEVIEW_URL'],
     ssl_enabled:             ENV['ONEVIEW_SSL_ENABLED'],
     log_level:               ENV['ONEVIEW_LOG_LEVEL'] || 'info',
-    api_version:             ENV['ONEVIEW_API_VERSION'] || 200,
+    api_version:             ENV['ONEVIEW_API_VERSION'] ? ENV['ONEVIEW_API_VERSION'].to_i : 200,
```

Comment:
@fgbulsoni 
When I was testing the manifests using env vars, I got this issue:

```
Error: /Stage[main]/Main/Oneview_fc_network[fc1]: Could not evaluate: comparison of String with 300 failed
```

The reason is because OneViewSDK is expecting api_version to be an Integer.

I've tested this solution using Linux, please can you run a manifest on Windows with the configuration stored in env vars to validate the solution?